### ISSUE TITLE
Pagination class

### ISSUE THREAD

**Issue by WanWizard** [fuel/core#601]
The status of the current Pagination class needs to be reviewed.

Does a bit of application-related code have a place in the core? If so, can we make it less bloated, and template driven so we can get rid of the HTML now generated by the class?

**Comment by jschreuder** [comment]
I'm in full agreement on this, less features - anything complex can go into extensions or the template.

**Comment by dhrrgn** [comment]
What if Pagination ::link() (I think that is the method). Took a callback which generates the links?  We call the callback for every link that needs to be generated.

**Comment by WanWizard** [comment]
Could be an option, but that requires quite a bit of logic that you need to store somewhere in your app, maybe we can do something clever with placeholders, and just pass a string?

And all HTML stuff has to go into a template, it doesn't belong in the class.

**Comment by Savageman** [comment]
Don't know if it will help, but I had to rewrite the pagination class to fit my needs.
My main concern was to not use a URI "segment" to store the page number.

The callback function is exactly what I did. The only parameter it takes is the current page number to generate the link for.

This way I could implement my app logic as I wanted:
- Store the page number somewhere in the URL in a way that fits our URL rewriting mechanism (not as an URI segment, nor as a GET parameter)
- Remove the page number from the URL for the first page
